### PR TITLE
Update agent

### PR DESCRIPTION
**Pull request by tutkat**
https://equinixjira.atlassian.net/browse/CXF-102608

Updated agent to identify as fabric-sdk-java

### CODE REVIEW

File: templates/Java/libraries/okhttp-gson/ApiClient.mustache [hunk 1]
```diff
@@ -267,7 +267,7 @@ public class ApiClient {
         json = new JSON();
 
         // Set default User-Agent.
-        setUserAgent("{{{httpUserAgent}}}{{^httpUserAgent}}OpenAPI-Generator/{{{artifactVersion}}}/java{{/httpUserAgent}}");
```
Comment (ctreatma) [inline templates/Java/libraries/okhttp-gson/ApiClient.mustache:270]: You don't need to modify the template for this.  You can set the HTTP user agent with a command-line flag for the generator.  Example from metal-java (the approach used there is the same as what is used in other Equinix SDKs as well): https://github.com/equinix-labs/metal-java/blob/main/Makefile#L52

Comment (tutkat) [reply templates/Java/libraries/okhttp-gson/ApiClient.mustache:270]: tanks for the tip :)

Comment (tutkat) [reply templates/Java/libraries/okhttp-gson/ApiClient.mustache]: @ctreatma  adjusted your recommendation

File: Makefile [hunk 2]
```diff
@@ -9,7 +9,9 @@ CURRENT_GID := $(shell id -g)
 
 # git repo : equinix fabric-java sdk
 GIT_ORG=equinix-labs
-GIT_REPO=fabric-java
+GIT_REPO=fabric-sdk-java
+
+PACKAGE_VERSION=$(shell cat spec/oas3.fabric.config.json | jq -r '.artifactVersion')
```
Comment (thogarty) [inline Makefile:14]: We need to remove `artifactVersion` and `artifactDescription` from `spec/oas3.fabric.config.json` file. They aren't being updated and they come directly from the OAS spec file.

Artifact version needs to come from a `version` file. See equivalent: https://github.com/equinix/equinix-sdk-go/blob/main/version

This gets read into PACKAGE_VERSION the following way: https://github.com/equinix/equinix-sdk-go/blob/c94de7bf1e1ffb039ad88ad7822a7518197a4b11/Makefile#L8C1-L8C37

We need to add a GHA job to update the `version` file. I believe the following job handles releases and updates to the version file: https://github.com/equinix/equinix-sdk-go/blob/main/.github/workflows/release.yaml

**Review by thogarty — CHANGES_REQUESTED** [review]
See comment. Needs update to process around versioning

**Comment by thogarty** [comment]
@tutkat , you're mixing too many different changes together. Don't update the test files in this PR. Just focus on the User Agent update.

Open a second PR for testing updates and provide context as to why they're needed.

**Comment by tutkat** [comment]
@thogarty corrected PR to make changes to change agent and versioning only based on another java repository which @ctreatma mentioned

**Review by thogarty — APPROVED** [review]
LGTM, @tutkat ! Thank you. Ship it.

**Pull request merged**